### DISCUSSION TITLE
GFM 与 Redcarpet 的不同点 — 码志

### DISCUSSION THREAD

**Discussion by mzlogin** [mzlogin/blog-comments#224] [Announcements]
https://mazhuang.org/2015/12/05/diff-between-gfm-and-redcarpet/

**Comment by adangran** [comment]
请教一个问题：你在本地如何预览？
我使用vim编辑markdown，之后用markdown-preview.vim或chrome的Markdown Preview Plus插件预览。问题是这两个的风格不太一样，markdown-preview.vim是GFM的，即行尾不需要留空格预览时能够自动换行，而Markdown Preview Plus则不会换行。用起来很痛苦，不知道是你如何做的。

**Comment by mzlogin** [comment]
那样并不能看到真实的效果，如果想看到真实的效果那就在本地搭建一个 Jekyll 环境来查看。

我现在并不在本地预览了，写好了源文件直接上传……这样做在开始的时候可能一篇文章会调整多次，但熟练之后就不怎么再有格式的问题了。

**Comment by adangran** [comment]
> @mzlogin
> 那样并不能看到真实的效果，如果想看到真实的效果那就在本地搭建一个 Jekyll 环境来查看。
> 
> 我现在并不在本地预览了，写好了源文件直接上传……这样做在开始的时候可能一篇文章会调整多次，但熟练之后就不怎么再有格式的问题了。

谢谢解答！

说到底还是对markdown解析不同造成的差异，看来在标准统一前是无解的了。

再请教一个关于vim-markdown-toc的问题。你可能是通过插件管理vim的插件的，而我是那种老式的vim用户，即安装插件是下载压缩包解压到vim的安装目录，我安装的一个markdown语法高亮插件https://github.com/plasticboy/vim-markdown/ 与vim-markdown-toc有两对文件是同名的：ftpdetect\markdown.vim和ftplugin\markdown.vim。我的解决方法是直接把这两个文件合并。但不知道是不是这样的原因， let g:vmt_auto_update_on_save = 1 好像不起作用，保存时并不会自动更新目录，这个有解吗？

**Comment by mzlogin** [comment]
可能是两个插件相互影响导致功能不可用了吧，可以调试一下代码看看，但我太懒不想弄 😢 

话说，

1. Vim 现在已经默认内置了 https://github.com/tpope/vim-markdown ，我觉得默认的 Markdown 语法高亮已经够用了呢，可以尝试直接看一下默认的效果；

2. 何不趁机尝试一下 Vundle / vim-plug / neobundle / dein.vim / Pathogen 等插件管理器呢，会方便很多，也方便不同机器之间的环境统一。比如我使用 Vundle 的配置 <https://github.com/mzlogin/config-files>

**Comment by elli999** [comment]
想问一下，Hexo博客框架用是的GFM还是Redcarpet啊？

**Comment by mzlogin** [comment]
> @elli999
> 想问一下，Hexo博客框架用是的GFM还是Redcarpet啊？

这个我不了解哈~没有使用过 Hexo。
